Client/src/utils/sorting-algo.ts

```diff
@@ -0,0 +1,61 @@
+export function bubbleSortAlgo(arr: number[]) {
+  const ArrayStates: number[][] = [];
+  let animations:number[][] = []
+  let hasChanged = true;
+  while (hasChanged) {
+    hasChanged = false;
+    for (let i = 0; i <= arr.length - 2; i++) {
+      if (arr[i] > arr[i + 1]) {
+        animations.push([i, arr[i], i+1, arr[i+1]]);
+        [arr[i], arr[i + 1]] = [arr[i + 1], arr[i]];
+        ArrayStates.push([...arr]);
+        hasChanged = true;
+      }
+    }
+  }
+  return {ArrayStates, animations};
+}
+
+export function insertionSortAlgo(arr: number[]) {
+  const ArrayStates: number[][] = [];
```

Comment:
This implementation can be improved, this is n^2 complexity

Reply:
This implementation is n log n
`function insertionSort(arr) {
  for (let i = 1; i < arr.length; i++) {
    let j = i;
    while (arr[j] < arr[j - 1] && j > 0) {
      [arr[j - 1], arr[j]] = [arr[j], arr[j - 1]];
      j--;
    }
  }
  return arr;
}`